### PR TITLE
Fix dockertest.py

### PR DESCRIPTION
**Pull request by cyli**
This moves everything from the misc directory to the buildscripts directory, and also fixes the downloader to work against the new .tgz files that docker distributes.

Also drop support for testing 1.8 and 1.9.

(the script runs successfully for me on master atm)

### CODE REVIEW

File: buildscripts/dockertest.py [hunk 1]
```diff
@@ -23,17 +24,13 @@
...
+DOCKERS = {}
 
 # delete any of these if you want to specify the docker binaries yourself
 DOWNLOAD_DOCKERS = {
-    "1.8": ("https://get.docker.com", "docker-1.8.3"),
-    "1.9": ("https://get.docker.com", "docker-1.9.1"),
-    "1.10": ("https://get.docker.com", "docker-1.10.3")
+    "1.10": ("https://get.docker.com", "docker-1.10.3"),
+    "1.11": ("https://get.docker.com", "docker-1.11.2"),
+    "1.12": ("https://get.docker.com", "docker-1.12.0"),
```
Comment (riyazdf) [inline buildscripts/dockertest.py]: 🎉

**Comment by riyazdf** [comment]
thank you for updating this, LGTM pending CI!

**Comment by endophage** [comment]
LGTM

**Comment by endophage** [comment]
Looks like Gordon is AFK. Merging without him.

**Pull request merged**